packages/compass-query-bar/src/stores/query-changed-store.js

```diff
@@ -81,8 +83,25 @@ const configureStore = (options = {}) => {
         } else {
           debug('Error: AppRegistry not available for query-changed-store');
         }
+
         if (globalRegistry) {
-          globalRegistry.emit('compass:query-bar:query-changed', newState);
+          const dataService = this.queryBarStore.dataService;
+          const nsobj = toNS(newState.ns);
+          let collectionType = '';
```

Comment:
we could try to stay away from mutation, especially in case where the mutation happens asychronously, it may seem obvious what is happening here right now but this is a super frequent reason for bugs.

as a suggestion for example, you could pull this out to a function:

```
if (globalRegistry) {
  globalRegistry.emit(
     'compass:query-bar:query-changed',
     {
       collectionType:  await getCollectionType(this.queryBarStore.dataService).catch(debug), 
       ...newState
     }); 
}
```

That's usually enough to make mutations not necessary:

```
function getCollectionType(dataService) {
   if (!dataService) { return; }
   const listCollections = util.promisify(dataService.listCollections.bind(dataService));
   const collection = await listCollections(nsobj.database, { name: nsobj.collection });
   return collection[0].options.timeseries ? 'time-series' : 'collection';
}
```